### ISSUE TITLE
GPU test failures

### ISSUE THREAD

**Issue by fvisin** [Theano/Theano#3733]
Some GPU test fail and are not currently run on Travis, so nobody noticed.

Specifically:
```
FAILED (SKIP=69, errors=2, failures=3)
FAIL: test_not_lazy_if_inplace (theano.tests.test_ifelse.test_ifelse)
FAIL: test_not_lazy_if_inplace (theano.sandbox.cuda.tests.test_opt.TestIfElse)
FAIL: theano.sandbox.cuda.tests.test_blas.test_downsample
ERROR: test_perform (theano.tensor.tests.test_slinalg.TestKron)
ERROR: theano.sandbox.cuda.tests.test_opt.test_opt_gpujoin_joinvectors_elemwise_then_minusone
```

Some extra information I gathered from @lamblin and @abergeron :
* `TestKron` is a known failure
* In `test_ifelse`, [this line](https://github.com/Theano/Theano/blob/cca20bafbb270679d9d06b808682ac99bae3ed9f/theano/tests/test_ifelse.py#L70) should be changed into:

        self.assertFunctionContains1(f, self.get_ifelse(1))
but it is not the only issue (the test still fails after fixing it).

**Comment by fvisin** [comment]
With the latest master this test does not fail anymore:
```FAIL: test_not_lazy_if_inplace (theano.tests.test_ifelse.test_ifelse)```

**Comment by nouiz** [comment]
The 3 first should be addressed by current  PR.

I'm checking the last one.

We have a daily buildbot, but I have difficulty to bring it back since the
change of the OS on that machine. This allow us to see the result of GPU
tests.

**Comment by lamblin** [comment]
testKron should be fixed by #3736

**Comment by nouiz** [comment]
They have been fixed and the daily buildbot is back. So I close this issue.
